core/datasource/external/ethcall/common/spec.go

```diff
@@ -46,14 +47,14 @@ type Spec struct {
 	L2ChainID             uint64
```

Comment:
I think instead of trying to pass a default chainID all the way through all these many, many, many, many, many, many layers of data-source types, we should just make this field `*uint64` and leave it as `nil` to implie its for the L1 chain. Then in `OnSpecActivated()` on the ethcall engine where we do this:
```
if e.chainID != d.L2ChainID {
    return nil
}
```

have something like

```
if d.L2ChainID == nil && e.iAmTheL1Engine {
    // this one is for me
   return
}

if e.chainID == d.L2ChainID {
	// I'm an L2 engine and this one is for me
        return
}

// this one is not for me
return nil
```

Reply:
Given that the L1 engine has a set ID, the check we have should have the same effect, surely

Reply:
It does, but has the advantage of not having to awkwardly pass a default chainID down all these functions that should really just be a basic type conversion.

There is also the second problem that if someone sets `nil` as the change ID when proposing a market for the field called `l2_chain_id`, when it comes out the other end in the event (and then through the API) the field called `l2_chain_id` will be populated with the Ethereum L1 chain ID, which doesn't make much sense.